cfme/cloud/instance/__init__.py

```diff
@@ -347,7 +307,53 @@ def power_control_from_cfme(self, *args, **kwargs):
                                        handle_alert=not kwargs.get('cancel', False))
 
 
-@navigator.register(Instance, 'All')
+@attr.s
+class InstanceCollection(VMCollection):
+    ENTITY = Instance
+
+    def create(self, vm_name, provider, form_values, cancel=False):
+        """Provisions an instance with the given properties through CFME
+
+        Args:
+            form_values: dictionary of form values for provisioning, structured into tabs
+
+        Note:
+            Calling create on a sub-class of instance will generate the properly formatted
+            dictionary when the correct fields are supplied.
+        """
+        instance = self.instantiate(vm_name, provider)
+        view = navigate_to(self, 'Provision')
+
+        # Only support 1 security group for now
+        # TODO: handle multiple
+        if ('environment' in form_values and
+                'security_groups' in form_values['environment'] and
+                isinstance(form_values['environment']['security_groups'], (list, tuple))):
+
+            first_group = form_values['environment']['security_groups'][0]
+            recursive_update(form_values, {'environment': {'security_groups': first_group}})
+
+        view.form.fill(form_values)
+
+        if cancel:
+            view.form.cancel_button.click()
+            # Redirects to Instance All
+            view = self.browser.create_view(InstanceAllView)
+            wait_for(lambda: view.is_displayed, timeout=10, delay=2, message='wait for redirect')
+            view.flash.assert_success_message(self.PROVISION_CANCEL)
+            view.flash.assert_no_error()
+        else:
+            view.form.submit_button.click()
+
+            view = self.appliance.browser.create_view(RequestsView)
+            wait_for(lambda: view.flash.messages, fail_condition=[], timeout=10, delay=2,
```

Comment:
**QUESTION:** Do we need the lambda here?

Reply:
Is a property not a callable?

Reply:
Property is callable, but we want to pass the bound property function to wait_for, and have it called on each iteration. If we just pass `wait_for(view.flash.messages, <etc>)` it will evaluate the property once, and pass its value to `wait_for`, instead of passing the bound function.

Wrapping in lambda prevents the property from being evaluated before being passed, and the lambda function itself is passed to wait_for and called on every iteration, evaluating the property value at that time.